File: Serene1/Serene1/Serene1.Web/Modules/Common/Imports/ServerTypings/Northwind.TerritoryForm.ts
﻿namespace Serene1.Northwind {
    export class TerritoryForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Territory';

    }

    export interface TerritoryForm {
        TerritoryID(): Serenity.StringEditor;
        TerritoryDescription(): Serenity.StringEditor;
        RegionID(): Serenity.LookupEditor;
    }

    [['TerritoryID', Serenity.StringEditor], ['TerritoryDescription', Serenity.StringEditor], ['RegionID', Serenity.LookupEditor]].forEach(x => TerritoryForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

File: Serene1/Serene1/Serene1.Script/Imports/ServerImports/Northwind.TerritoryForm.cs
﻿using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene1.Northwind
{
    public partial class TerritoryForm : PrefixedContext
    {
        [InlineConstant]
        public const string FormKey = "Northwind.Territory";

        public TerritoryForm(string idPrefix) : base(idPrefix) { }

        public StringEditor TerritoryID { [InlineCode("{this}.w('TerritoryID', Serenity.StringEditor)")] get; private set; }
        public StringEditor TerritoryDescription { [InlineCode("{this}.w('TerritoryDescription', Serenity.StringEditor)")] get; private set; }
        public LookupEditor RegionID { [InlineCode("{this}.w('RegionID', Serenity.LookupEditor)")] get; private set; }
    }
}

